### DISCUSSION TITLE
[Bug] java-agent 8.7+,TraceContext.traceId() can't get traceId in ServerInterceptor

### DISCUSSION THREAD

**Discussion by zzhi** [apache/skywalking#9087] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

with skywalking-agent 8.7+ ,  TraceContext.traceId() can't get traceId in ServerInterceptor

skywalking-agent 8.6 ok 

![AK0IGRoc8y](https://user-images.githubusercontent.com/10458695/168822031-6ae7d87d-099d-4d3f-8b03-a4b59c6e9ed2.png)

skywalking-agent 8.7  tid:"N/A"
![1Oiti7QE2L](https://user-images.githubusercontent.com/10458695/168822218-942f6232-107f-4d12-ae84-8c3daf5aa3ce.png)

### What you expected to happen

with skywalking-agent 8.7、8.8+ ,  TraceContext.traceId() can get traceId in ServerInterceptor

### How to reproduce

-javaagent:/Users/bj89089ml/Documents/sky-agent/apache-skywalking-apm-bin-es7-8.7/agent/skywalking-agent.jar
-Dskywalking.agent.service_name=test
-Dskywalking.collector.backend_service=127.0.0.1:11800
-Dskywalking.logging.output=CONSOLE

 maven dependency apm-toolkit-trace version 8.7.0+
         

grpc-scenario
code in org.apache.skywalking.apm.testcase.grpc.provider.interceptor.ProviderInterceptor

### Anything else

no

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
The upstream doesn't guarantee all plugins' behaviors didn't change from version to version. We also don't think this is a bug. The current version is on 8.11, there are several changes that happened for the gRPC plugin, you should track them to see what changed.

**Comment by andotorg** [comment]
Check to see if this question is helpful to you？
https://github.com/apache/skywalking/discussions/9113
